common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -4861,6 +4861,26 @@ static jobjectArray NativeCrypto_get_X509_GENERAL_NAME_stack(JNIEnv* env, jclass
     return joa.release();
 }
 
+/*
+ * Checks an ASN1_TIME for validity, converts to epoch time in milliseconds.
+ */
+static jlong ASN1_TIME_check_and_convert_to_posix(JNIEnv* env, const ASN1_TIME* time) {
+    if (!ASN1_TIME_check(time)) {
+        JNI_TRACE("ASN1_time_check_and_convert(%p) => Invalid date format", time);
+        conscrypt::jniutil::throwParsingException(env, "Invalid date format");
+        return 0;
+    }
+
+    int64_t retval;
+    if (!ASN1_TIME_to_posix(time, &retval)) {
+        JNI_TRACE("ASN1_time_check_and_convert(%p) => Invalid date value", time);
+        conscrypt::jniutil::throwParsingException(env, "Invalid date value");
+        return 0;
+    }
+
+    return static_cast<jlong>(retval * 1000);
```

Comment:
This would be UB if it overflowed, but it won't:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/boringssl/src/crypto/asn1/posix_time.c;drc=6cda656e4a2e1545e23b30dfb7dc05fe423e3f6c;l=76

Maybe worth a comment that `ASN1_TIME_to_posix` can only return times from 0000 to 9999, so the millisecond count fits in `int64_t`.

Reply:
Done.

Feels like we should have some test coverage for this, but I couldn't persuade openssl to generate a test certificate far enough in the future... Will look into it for.a future CL.